File: python/svix/models/message_out.py
# this file is @generated
import typing as t
from datetime import datetime

from .common import BaseModel


class MessageOut(BaseModel):
    channels: t.Optional[t.List[str]] = None
    """List of free-form identifiers that endpoints can filter by"""

    event_id: t.Optional[str] = None
    """Optional unique identifier for the message"""

    event_type: str
    """The event type's name"""

    id: str
    """The Message's ID."""

    payload: t.Dict[str, t.Any]

    tags: t.Optional[t.List[str]] = None

    timestamp: datetime

File: rust/src/models/message_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct MessageOut {
    /// Optional unique identifier for the message
    #[serde(rename = "eventId")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub event_id: Option<String>,

    /// The event type's name
    #[serde(rename = "eventType")]
    pub event_type: String,

    pub payload: serde_json::Value,

    /// List of free-form identifiers that endpoints can filter by
    #[serde(skip_serializing_if = "Option::is_none")]
    pub channels: Option<Vec<String>>,

    /// The Message's ID.
    pub id: String,

    pub timestamp: String,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub tags: Option<Vec<String>>,

    #[serde(rename = "deliverAt")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub deliver_at: Option<String>,
}

impl MessageOut {
    pub fn new(
        event_type: String,
        payload: serde_json::Value,
        id: String,
        timestamp: String,
    ) -> Self {
        Self {
            event_id: None,
            event_type,
            payload,
            channels: None,
            id,
            timestamp,
            tags: None,
            deliver_at: None,
        }
    }
}
